packages/components/src/navigation/stories/index.js

```diff
@@ -1,88 +1,103 @@
+/**
+ * WordPress dependencies
+ */
+import { useState } from '@wordpress/element';
+
 /**
  * Internal dependencies
  */
 import Navigation from '../';
+import NavigationBackButton from '../back-button';
+import NavigationMenu from '../menu';
+import NavigationTitle from '../title';
 
 export default {
 	title: 'Components/Navigation',
 	component: Navigation,
 };
 
 const data = [
-	{ title: 'WooCommerce', slug: 'root', back: 'Dashboard' },
-	{ title: 'Home', slug: 'home', parent: 'root', menu: 'primary' },
+	{ title: 'Home', id: 'home' },
 	{
 		title: 'Analytics',
-		slug: 'analytics',
-		parent: 'root',
-		back: 'WooCommerce Home',
-		menu: 'primary',
+		id: 'analytics',
 	},
 	{
 		title: 'Orders',
-		slug: 'orders',
-		parent: 'root',
-		back: 'WooCommerce Home',
-		menu: 'primary',
+		id: 'orders',
 	},
 	{
 		title: 'Overview',
-		slug: 'overview',
+		id: 'overview',
 		parent: 'analytics',
 	},
 	{
 		title: 'Products report',
-		slug: 'products',
+		id: 'products',
 		parent: 'analytics',
 	},
 	{
 		title: 'All orders',
-		slug: 'all_orders',
+		id: 'all_orders',
 		parent: 'orders',
 	},
 	{
 		title: 'Payouts',
-		slug: 'payouts',
+		id: 'payouts',
 		parent: 'orders',
 	},
 	{
 		title: 'Settings',
-		slug: 'settings',
-		parent: 'root',
-		back: 'WooCommerce Home',
+		id: 'settings',
 		menu: 'secondary',
 	},
 	{
 		title: 'Extensions',
-		slug: 'extensions',
-		parent: 'root',
-		back: 'WooCommerce Home',
+		id: 'extensions',
 		menu: 'secondary',
 	},
 	{
 		title: 'General',
-		slug: 'general',
+		id: 'general',
 		parent: 'settings',
 	},
 	{
 		title: 'Tax',
-		slug: 'tax',
+		id: 'tax',
 		parent: 'settings',
 	},
 	{
 		title: 'My extensions',
-		slug: 'my_extensions',
+		id: 'my_extensions',
 		parent: 'extensions',
 	},
 	{
 		title: 'Marketplace',
-		slug: 'marketplace',
+		id: 'marketplace',
 		parent: 'extensions',
 	},
 ];
 
 function Example() {
-	return <Navigation data={ data } initial="home" />;
+	const [ active, setActive ] = useState( 'home' );
+	const items = data.map( ( item ) => {
+		item.onClick = () => setActive( item.id );
+		return item;
+	} );
+
+	return (
+		<Navigation active={ active } items={ items }>
```

Comment:
Instead of passing an array of `items` here I'd prefer to have the consumer iterate over the items and create navigation items:

```js
{ items.map( item => <NavigationMenuItem>{ item.title }</NavigtionMenuItem> }
```

However, this means the top level `Navigation` component does not have access to the full array and can't parse information needed for the title and back button components.

Reply:
>However, this means the top level Navigation component does not have access to the full array and can't parse information needed for the title and back button components.

One way to get around this can be to have `<Navigation />` accept a single child that is a function. This way, any parameters can be supplied as function arguments. Something along these lines?

```js
<Navigation active={ active } items={ data }>
	{ ( { title, parent } ) => (
		<>
			<NavigationBackButton parent={ parent } />
			<NavigationTitle title={ title } />
			<NavigationMenu>
				{ data.map( ( item ) => (
					<NavigationMenuItem
						key={ item.id }
						item={ item }
						isActive={ item.id === active }
					/>
				) ) }
			</NavigationMenu>
		</>
	) }
</Navigation>
```
See the Tab Panel as an example:
https://github.com/WordPress/gutenberg/tree/master/packages/components/src/tab-panel